z2/src/node_spec.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::default::Default;
+use std::fmt;
+use zilliqa::range_map::RangeMap;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct NodeDesc {
+    pub is_validator: bool,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct Composition {
+    pub nodes: HashMap<u64, NodeDesc>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct NodeSpec {
```

Comment:
I wonder if `NetworkSpec` or `NodesSpec` is more clear here

Reply:
I'll leave that to the language lawyers - when those who care enough have decided what they want, very happy to implement it :-)